src/routes/webhooks.ts

```diff
@@ -45,6 +51,32 @@ const verifyRequest = (publicKey: string, req: FastifyRequest): boolean => {
   );
 };
 
+const verifyCIOSignature = (
+  webhookSigningSecret: string,
+  req: FastifyRequest,
+): boolean => {
+  const timestamp = req.headers['x-cio-timestamp'] as string;
+  const signature = req.headers['x-cio-signature'] as string;
+
+  if (!timestamp || !signature) {
+    return false;
+  }
+
+  const hmac = createHmac('sha256', webhookSigningSecret);
+  hmac.update(`v0:${timestamp}:`);
+  hmac.update(req.rawBody);
+
+  const hash = hmac.digest();
+
+  if (!timingSafeEqual(hash, Buffer.from(signature, 'hex'))) {
+    logger.debug("Signature didn't match");
+    return false;
+  }
+
+  logger.debug('Signature matched!');
```

Comment:
I know it is just debug logging, but probably something like "CIO signature matched"? Same above.

Reply:
Added